### DISCUSSION TITLE
Problem of installation

### DISCUSSION THREAD

**Discussion by j83gros** [thelfer/tfel#451] [Q&A]
Hello,
I am a student currently in an intership and my purpose is to implement a mechanical law in FRAPCON. Previously, I have to implement and test it using the MFront tool.
However, I have encountered several problems when I tried to install MFront. I susupect it could come from my own computer but I have praticly never used Linux interface before so I may just not understand the problem. Following is a list of all the problem I have encountered and they final issue.

First I tried to use Msys and the “.bat”  files. I didn’t succeed to use Cmake with the first one, even if I already have it installed and the “.bat” files don’t run themselves.
So, I tried to setup a virtual machine (namely Ubuntu), which I had already done in prepa, but I encountered some problems I don’t know how to resolve. The main problem is that it seems I can’t fully perform the first update nor install some important applications as g++ and gcc. I tried to override the problem by using another C/C++ implementation, but I am not sure it works because anyway, I have never been able to implement CMake or to clone the MFront files on my virtual machine: there is always a problem like: 
![image](https://github.com/thelfer/tfel/assets/151615377/8dbb9da9-377d-48b8-9b36-34f5c8a5419e)
 

After that, I tried another option which was to run everything directly with visual studio.
Two lines bothered me, but I didn’t know if it was normal so I tried to compile it nevertheless:
 
![image](https://github.com/thelfer/tfel/assets/151615377/f1955e6f-6d8b-491a-93b4-60b90077551a)

It returns me an error and a warning:
 
![image](https://github.com/thelfer/tfel/assets/151615377/03ed7192-b43a-4fa9-adc9-b9725267dd71)

I must precise that, at this point, I didn’t change the configuration of MFront: before setting it up to my wish, I wanted to be sure it could run with the default configuration so I assume the error not really comes from the code and can come from my own configuration of Cmake or something else.

Has anyone advice to help me ?
Thank you for your attention,
Julien Gros

**Comment by thelfer** [comment]
@j83gros Thanks for your interest in `MFront`. This is very interesting to link `MFront` and `Frapcon`. Do not hesitate to ask for help.

Can you precise a little bit the context:

- what compiler do you use to compile MFront
- how do you plan to use MFront law in `Frapcon`:
  -  Which interface did you plan to use (`generic` ?).
  -  Do you want to introduce a single law or have something very generic ?  In the second case, did you consider using `MGIS` ?

Concerning your troubles, Version 4.x is currently not compatible with Visual Studio. You shall use either `mingw` or the `intel one api`. You may also consider using version 3.4.x.

> So, I tried to setup a virtual machine (namely Ubuntu), which I had already done in prepa, but I encountered some problems I don’t know how to resolve. The main problem is that it seems I can’t fully perform the first update nor install some important applications as g++ and gcc. I tried to override the problem by using another C/C++ implementation, but I am not sure it works because anyway, I have never been able to implement CMake or to clone the MFront files on my virtual machine

What virtualizer did you use ? Virtualbox ? If so, you may want to install a new image from scratch.

On windows, you may want to use WSL which I know to work flawlessly.

**Reply by j83gros** [reply]
Thanks for your answer.
I used the clang compiler but I did not thought to try another one.
I prefer using the python interface. 
Concerning the law: it is a single law but with two sets of parameters so I did not consider to use MGIS.
I think that I used WSl as a virtualizer.
I will try to install the version 3.4.x.

**Reply by thelfer** [reply]
> I think that I used WSl as a virtualizer.

So I can't understand why you have troubles. Maybe shall you try to reinstall your ubuntu images from scratch.

> I prefer using the python interface.

The python interface is only available for material properties, not for behaviours. I am a bit confused

> I will try to install the version 3.4.x.

In an Ubuntu image, that would not help. Version 3.4.x is only required if you use Visual Studio.

**Reply by j83gros** [reply]
> The python interface is only available for material properties, not for behaviours. I am a bit confused

Ok, it may just be me which did not see the difference, I will search for another one.

> In an Ubuntu image, that would not help. Version 3.4.x is only required if you use Visual Studio.

I will first try to use Visual studio and the 3.4.x version, I think it would be sufficient for what I have to do.

> So I can't understand why you have troubles. Maybe shall you try to reinstall your ubuntu images from scratch.

I will try this if I need it, thank you.

**Reply by thelfer** [reply]
> Ok, it may just be me which did not see the difference, I will search for another one.

Material properties are things like Young's modulus, Poisson's ratio, etc...

Behaviours describes the mechanical response in terms of stress and the evolution of some internal state variables associated with damage, plasticity, etc..

> I will first try to use Visual studio and the 3.4.x version, I think it would be sufficient for what I have to do

I really don't get what you want to do. Do you want to compile TFEL on Linux or on Windows ?

You seem to want to use Ubuntu, so I don't get the point of using Visual Studio which is (mostly) Windows-centric.

**Reply by j83gros** [reply]
> Do you want to compile TFEL on Linux or on Windows ?

Sorry, I want to compile TFEl on Windows. Linux was more an alternative solution.
This is why I will use Visual Studio and the 3.4.. version.

**Comment by fpguillet** [comment]
I've compiled TFEL on Windows using Visual Studio and Intel classic compilers. It's not straightforward and I had to modify one or two (can't remember) Visual Studio header files. The only reason you'd want to go through this pain is you you want to provide MFront behaviors to a Windows code through an MFront library. Otherwise, it's much, much simpler to setup WSL and install TFEL. The reason WSL exist is because a certain number of tools are Linux only and porting them is a real hassle. With WSL you can call Linux tools directly from within Windows. In that respect, WSL is not a Virtual Machine, for example it has access to Windows filesystem.

This probably means you have to learn a few Linux basics and tricks, but in the scientific computation world Unix rules, so maybe it's the right time to expand your skills.

________________________________
De : j83gros ***@***.***>
Envoyé : mardi 21 novembre 2023 12:25:57
À : thelfer/tfel
Cc : Subscribed
Objet : Re: [thelfer/tfel] Problem of installation (Discussion #451)


Do you want to compile TFEL on Linux or on Windows ?

Sorry, I want to compile TFEl on Windows. Linux was more an alternative solution.
This is why I will use Visual Studio and the 3.4.. version.

**Reply by j83gros** [reply]
I understand your point and have never seen WSL this way but what you say makes sense.
It is true that I will surely need this sort of skills later, you have convinced  me to use a wsl, thank you.

**Reply by thelfer** [reply]
There could one reason to favor Windows build: this reason would be that Frapcon is only available on Windows. Otherwise go ahead on use Linux !

**Reply by thelfer** [reply]
@fpguillet I just checked: after two small fixes both rliv-3.3 and rliv-3.4 compiles fine with Visual Studio.
Versions 4.x are hopeless.
I will try to support Visual Studio in the upcoming 5.0 branch

**Comment by fpguillet** [comment]
@thelfer

If you support Visual Studio (or rather the MSVC distribution which was available for free as a separate installer but it's no longer the case), check also Intel OneApi which is now available for free if I'm correct. Intel compiler is almost manadatory if you need to compile FORTRAN code. Well there's also MinGW32 but I admit I never could install it properly.

________________________________
De : Thomas Helfer ***@***.***>
Envoyé : mercredi 22 novembre 2023 11:01:29
À : thelfer/tfel
Cc : GUILLET François 142817; Mention
Objet : Re: [thelfer/tfel] Problem of installation (Discussion #451)


@fpguillet<https://github.com/fpguillet> I just checked: after two small fixes both rliv-3.3 and rliv-3.4 compiles fine with Visual Studio.
Versions 4.x are hopeless.
I will try to support Visual Studio in the upcoming 5.0 branch

**Reply by thelfer** [reply]
I'll try Intel One API ASAP.
We support `mingw` for years with no real problems.
I'll give also `clang` a try
